### PR TITLE
Merge basic-page-structure branch into development branch

### PR DESCRIPTION
**Pull request by aristides1000**
the following activities were carried out:

- [x] IMPORTANT NOTE This is the first time you will use Gitflow. Keep that in mind!
- [x] Create an npm project with webpack. (If you need a refresher, check the previous project or webpack exercise).
- [x] Write plain HTML markup with minimum styling (just to make the layout), because you will implement styles in the following steps.
- [x] Don't make API calls (you'll do it in a following lesson).
- [x] Use ES6 modules, with import and export.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/* eslint-disable linebreak-style */
+/* eslint-disable no-unused-vars */
+import _ from 'lodash';
+import './style.css';
+
+function component() {
+  const element = document.createElement('div');
+
+  return element;
+}
```
Comment (oxenprogrammer) [inline src/index.js]: **Changes Required**
- [x] The use of `arrow function` is preferred to the `function` keyword. Let us refactor this into an ES6 arrow function. You can read more [here](https://zendev.com/2018/10/01/javascript-arrow-functions-how-why-when.html) on why you should prefer ES6 arrow functions to the `function keyword`.

File: src/index.html [hunk 2]
```diff
@@ -0,0 +1,63 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.0/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-KyZXEAg3QhqLMpG8r+8fhAXLRk2vvoC2f3B09zVXn8CA5QIVfZOJ3BCsw2P0p/We" crossorigin="anonymous">
+  <title>Wbpack Exercise</title>
+</head>
+<body>
+  <div class="container">
```
Comment (oxenprogrammer) [inline src/index.html:10]: **Changes Required**
- [x] The `index.html` should only contain the root element, i.e `.container`. The rest of the html should be generated dynamically by manipulating the `DOM` using JavaScript. You could checkout [this freecodecamp](https://www.freecodecamp.org/news/how-to-manipulate-the-dom-beginners-guide/) article plus any other web resources in line with this subject. Please let us refactor this accordingly.

**Review by oxenprogrammer — CHANGES_REQUESTED** [review]
Hi @aristides1000 , 

Good job so far!
I can see you're really putting effort to become a world-class software developer :+1:
There are some issues that you still need to address in order to move forward to the next project.
Trust me you got this! :muscle:


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Consider the optional changes nice to have features and only work on them when you have satisfied the required changes_

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/structure.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+/* eslint-disable linebreak-style */
+
+const structure = () => {
+  const container = document.querySelector('.container');
+
+  container.innerHTML = `<div class="row">
+  <div class="col-12 mt-5">
```
Comment (oxenprogrammer) [inline src/structure.js]: **Changes Required**
- [x] The html is hard coded here. Let us use JavaScript to dynamically create the html on the `DOM`. 
Let us try to make every HTML element in the DOM dynamically generated.
Here is an example code snippet on how you could generate some html using JavaScript.

```
 const bookContainer = document.createElement('div');
    bookContainer.setAttribute('class', 'bookContainer');
    bookContainer.setAttribute('id', '');
    bookShelf.appendChild(bookContainer);

    const bookTitle = document.createElement('p');
    bookTitle.textContent = book.title;

    const bookPages = document.createElement('p');
    bookPages.textContent = book.pages;

    const bookAuthor = document.createElement('p');
    bookAuthor.textContent = book.author;

    bookContainer.appendChild(bookTitle);
    bookContainer.appendChild(bookPages);
    bookContainer.appendChild(bookAuthor);
    ```

**Review by oxenprogrammer — CHANGES_REQUESTED** [review]
Hi @aristides1000 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet. :muscle:


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Consider the optional changes nice to have features and only work on them when you have satisfied the required changes_

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/structure.js (hunk 3 above)
Comment (aristides1000) [reply src/structure.js]: Thank you for your correction, the requested changes have already been implemented

**Comment by aristides1000** [comment]
The following requested implementations were made in this fix:

- [x] Make every HTML element in the DOM dynamically generated.

**Review by carlos-ssh — APPROVED** [review]
Hi @aristides1000 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**